### ISSUE TITLE
Mutation conventions input default values not respected

### ISSUE THREAD

**Issue by tnc1997** [ChilliCream/graphql-platform#6691]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Product

Hot Chocolate

### Describe the bug

It appears that when using mutation conventions, input default values are not being respected.

When creating the input class manually input default values behave as expected.

I would expect the input class from mutation conventions to respect any default values.

### Steps to reproduce

```csharp
var result =
    await new ServiceCollection()
        .AddGraphQL()
        .AddQueryType(d => d.Field("abc").Resolve("def"))
        .AddMutationType<MutationWithMutationConventionsWithDefaultValue>()
        .AddMutationConventions()
        .ExecuteRequestAsync(
            """
            mutation {
                doSomething(input: { }) {
                    boolean
                }
            }
            """);

result.MatchInlineSnapshot(
    """
    {
      "data": {
        "doSomething": {
          "boolean": true
        }
      }
    }
    """);
```

```csharp
public class MutationWithMutationConventionsWithDefaultValue
{
    public bool DoSomething(bool something = true)
        => something;
}
```

### Relevant log output

```shell
{
  "errors": [
    {
      "message": "Cannot accept null for non-nullable input.",
      "path": [
        "something"
      ],
      "extensions": {
        "code": "HC0018"
      }
    }
  ]
}
```

### Version

13.7.0

**Comment by tnc1997** [comment]
I have added tests to the `HotChocolate.Core` solution that reproduce this issue [here](https://github.com/tnc1997/graphql-platform/commit/955650503dcac38da1b69c782d24d4291dc0f5cf).

I have also started working on a (naive) change to the `MutationConventionMiddleware` that fixes this issue [here](https://github.com/tnc1997/graphql-platform/commit/cb12cb128b0096fc8e67167dc025808e8a6ff26e).

**Comment by michaelstaib** [comment]
This is not an issue of the middleware I think.

**Comment by tnc1997** [comment]
> This is not an issue of the middleware I think.

That was the first place I started looking and noticed that `input` contains the default value but `inputLiteral` does not.

**Comment by michaelstaib** [comment]
This is fixed with the next 13.8 preview build ... thanks for reporting/
